### PR TITLE
Setup PHPCS and cleanup linter configuration

### PR DESCRIPTION
**Pull request by virtualtam**
This is a first step towards adding linters to the development flow (Make, Travis).

It fixes most trivial PSR-1 and PSR-2 warnings, with the remaining errors belonging to these categories (related to autoloading):
- namespacing;
- one class per source file;
- PHP files should either declare symbols, or execute logic with effects.

---

Relates to #95

Added:
- Setup [PHP Code Sniffer](https://github.com/squizlabs/PHP_CodeSniffer) (phpcs) for [PSR-1](https://www.php-fig.org/psr/psr-1/) and [PSR-2](https://www.php-fig.org/psr/psr-2/)

Fixed:
- Fix spacing and indentation warnings and errors with PHP Code BeautiFier (phpcbf)
- Fix line-length warnings

Removed:
- PHP Copy Paste Detector dev-dependency and Make targets
- PHP Mess Detector dev-dependency and Make targets

### CODE REVIEW

File: application/FeedBuilder.php [hunk 1]
```diff
@@ -163,7 +163,8 @@ protected function buildItem($link, $pageaddr)
             $upDate = $link['updated'];
             $link['up_iso_date'] = $this->getIsoDate($upDate, DateTime::ATOM);
         } else {
-            $link['up_iso_date'] = $this->getIsoDate($pubDate, DateTime::ATOM);;
+            $link['up_iso_date'] = $this->getIsoDate($pubDate, DateTime::ATOM);
+            ;
```
Comment (ArthurHoaro) [inline application/FeedBuilder.php]: I think it can safely be removed :)

File: index.php [hunk 2]
```diff
@@ -1260,11 +1272,13 @@ function renderPage($conf, $pluginManager, $LINKSDB, $history, $sessionManager,
...
-    if (isset($_GET['edit_link']))
-    {
+    if (isset($_GET['edit_link'])) {
         $id = (int) escape($_GET['edit_link']);
         $link = $LINKSDB[$id];  // Read database
-        if (!$link) { header('Location: ?'); exit; } // Link not found in database.
+        if (!$link) {
+            header('Location: ?');
+            exit;
+        } // Link not found in database.
```
Comment (ArthurHoaro) [inline index.php]: Comment before.

File: index.php [hunk 3]
```diff
@@ -1739,16 +1761,19 @@ function buildLinkList($PAGE, $LINKSDB, $conf, $pluginManager, $loginManager)
...
+    }
 
 
     // This part makes sure sessions works correctly.
     // (Because on some hosts, session.save_path may not be set correctly,
     // or we may not have write access to it.)
-    if (isset($_GET['test_session']) && ( !isset($_SESSION) || !isset($_SESSION['session_tested']) || $_SESSION['session_tested']!='Working'))
-    {
+    if (isset($_GET['test_session'])
+        && ( !isset($_SESSION) || !isset($_SESSION['session_tested']) || $_SESSION['session_tested']!='Working')) {
```
Comment (ArthurHoaro) [inline index.php]: New lines and space after the `(`

**Comment by virtualtam** [comment]
Thanks for the review! PR updated

**Comment by virtualtam** [comment]
@nodiscc I'm not sure for now, as if the linter is non-blocking, Travis job output will rarely (never?) be consulted...

I've started working towards improving resource namespacing, a task that proved to be far from trivial (kinda related to the mixed use of classes and global functions). I'll try to submit a PR soon so we can discuss the best approach to adopt for refactoring the app into proper namespaces/packages :)

**Pull request merged**